src/app/util/af-types.h

```diff
@@ -764,17 +767,17 @@ typedef struct
 #endif // ZCL_USING_COLOR_CONTROL_CLUSTER_SERVER
 #ifdef ZCL_USING_DOOR_LOCK_CLUSTER_SERVER
     bool hasLockStateValue;
-    uint8_t lockStateValue;
+    chip::app::DataModel::Nullable<chip::app::Clusters::DoorLock::DlLockState> lockStateValue;
 #endif
 #ifdef ZCL_USING_WINDOW_COVERING_CLUSTER_SERVER
     bool hasCurrentPositionLiftPercentageValue;
-    uint8_t currentPositionLiftPercentageValue;
+    chip::app::DataModel::Nullable<chip::Percent> currentPositionLiftPercentageValue;
     bool hasCurrentPositionTiltPercentageValue;
-    uint8_t currentPositionTiltPercentageValue;
+    chip::app::DataModel::Nullable<chip::Percent> currentPositionTiltPercentageValue;
     bool hasTargetPositionLiftPercent100thsValue;
-    uint16_t targetPositionLiftPercent100thsValue;
+    chip::app::DataModel::Nullable<chip::Percent100ths> targetPositionLiftPercent100thsValue;
     bool hasTargetPositionTiltPercent100thsValue;
-    uint16_t targetPositionTiltPercent100thsValue;
+    chip::app::DataModel::Nullable<chip::Percent100ths> targetPositionTiltPercent100thsValue;
```

Comment:
can we use definitions done into the WindowCovering server location here or this is not possible !

Reply:
I'm not sure.  Some people are not happy with including cluster-objects.h in public API headers...

Filed https://github.com/project-chip/connectedhomeip/issues/24660 to track this.